application-admintools-default/src/main/java/com/xwiki/admintools/internal/security/EntityRightsProvider.java

```diff
@@ -0,0 +1,235 @@
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.query.SecureQuery;
+import org.xwiki.search.solr.SolrUtils;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.admintools.security.RightsResult;
+
+/**
+ * Retrieve data about the allowed/denied rights for a certain entity type.
+ *
+ * @version $Id$
+ * @since 1.2
+ */
+@Component(roles = EntityRightsProvider.class)
+@Singleton
+public class EntityRightsProvider extends AbstractRightsProvider
+{
+    private static final String XWIKI = "XWiki";
+
+    private static final String XWIKI_RIGHTS = "XWikiRights";
+
+    private static final String XWIKI_SERVER_PREFIX = "XWikiServer";
+
+    private static final String SPACE_TYPE = "Space";
+
+    private static final String PAGE_TYPE = "Page";
+
+    private static final String GLOBAL_TYPE = "Global";
+
+    private static final String WIKI_KEY = "wikiName";
+
+    private static final LocalDocumentReference GLOBAL_RIGHTS_CLASS =
+        new LocalDocumentReference(XWIKI, "XWikiGlobalRights");
+
+    private static final LocalDocumentReference DOCUMENT_RIGHTS_CLASS = new LocalDocumentReference(XWIKI, XWIKI_RIGHTS);
+
+    @Inject
+    private DocumentReferenceResolver<SolrDocument> solrDocumentReferenceResolver;
+
+    @Inject
+    @Named("secure")
+    private QueryManager secureQueryManager;
+
+    @Inject
+    private SolrUtils solrUtils;
+
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> documentReferenceResolver;
+
+    /**
+     * Retrieves a filtered and sorted list of {@link RightsResult} representing the rights for the given parameters.
+     *
+     * @param filters a map of filters to apply.
+     * @param sortColumn the column used for sorting.
+     * @param order the sorting order (asc or desc).
+     * @param entityType the type of entity for which rights are retrieved.
+     * @return a filtered and sorted {@link List} of {@link RightsResult}.
+     */
+    public List<RightsResult> getEntityRights(Map<String, String> filters, String sortColumn, String order,
+        String entityType)
+    {
+        try {
+            Set<RightsResult> rightsResults = new HashSet<>();
+            switch (filters.get(TYPE_KEY) == null ? "" : filters.get(TYPE_KEY)) {
```

Comment:
I would add a small comment to explain what global right are compared to the others